### ISSUE TITLE
[BUG] Not able to create VirtualMachineScaleSetVM instance for unit test. Ambiguous constructor call for Azure compute SDK > 47.0

### ISSUE THREAD

**Issue by KalpeshChavan12** [Azure/azure-sdk-for-net#23359]
**Describe the bug**
Compilation error while creating instance of VirtualMachineScaleSetVM class instance for unit test.

**Expected behavior**
Following should compile
```cs
 var vmssVm = new VirtualMachineScaleSetVM(location: "",
                                                    sku: new Sku() { Name = "xyz" },
                                                    storageProfile: new StorageProfile()
                                                    {
                                                        OsDisk = new OSDisk()
                                                        {
                                                            OsType = OperatingSystemTypes.Linux
                                                        }
                                                    });

```
**Actual behavior (include Exception or Stack Trace)**
What is the actual behavior?
Compilation error, ambiguous call
```

Program.cs(46,30,46,54): error CS0121: The call is ambiguous between the following methods or properties: 'VirtualMachineScaleSetVM.VirtualMachineScaleSetVM(string, string, string, string, IDictionary<string, string>, string, Sku, bool?, string, VirtualMachineScaleSetVMInstanceView, HardwareProfile, StorageProfile, AdditionalCapabilities, OSProfile, SecurityProfile, NetworkProfile, VirtualMachineScaleSetVMNetworkProfileConfiguration, DiagnosticsProfile, SubResource, string, string, string, VirtualMachineScaleSetVMProtectionPolicy, Plan, IList<VirtualMachineExtension>, IList<string>)' and 'VirtualMachineScaleSetVM.VirtualMachineScaleSetVM(string, string, string, string, IDictionary<string, string>, string, Sku, bool?, string, VirtualMachineScaleSetVMInstanceView, HardwareProfile, StorageProfile, AdditionalCapabilities, OSProfile, SecurityProfile, NetworkProfile, VirtualMachineScaleSetVMNetworkProfileConfiguration, DiagnosticsProfile, SubResource, string, string, string, VirtualMachineScaleSetVMProtectionPolicy, Plan, IList<VirtualMachineExtension>)'
```

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Create new project and create class as mentioned in issue.
2.  Use https://www.nuget.org/packages/Microsoft.Azure.Management.Compute >= 47.0
2. Build project

Note: Build successful version <= 46.0.0

**Environment:**
 - Name and version of the Library package used: [e.g. Azure.Storage.Blobs 12.2.0] 
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): [e.g. Azure AppService or Windows 10 .NET Framework 4.8]
 - IDE and version ::  Visual Studio 19

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by yungezz** [comment]
route to service team to have a look

**Comment by amjads1** [comment]
@avirishuv Could you please look into the issue?

**Comment by avirishuv** [comment]
hi @yungezz is this an issue in the SDK layer, would you mind investigating?

**Comment by yungezz** [comment]
hi @HarveyLink could you help to take a look? thanks

**Comment by HarveyLink** [comment]
Hi @KalpeshChavan12 , this issue has been fixed in version 49.1.0, would you mind update to the latest version and try your code?
